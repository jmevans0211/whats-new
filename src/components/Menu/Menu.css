ul {
  border-right: black solid 5px;
  font-size: 26px;
  margin: 0;
  width: 20%;
}

li {
  cursor: pointer;
  margin: 30px 0;
}