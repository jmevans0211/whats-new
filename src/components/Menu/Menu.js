import React from 'react';
import './Menu.css'

const Menu = (props) => {

  const handleClick = (event) => {
    console.log(event.target.className)
    props.selectTopic(event.target.className)
  }

  return (
    <ul>
      <li onClick={handleClick} className="entertainment">Entertainment</li>
      <li onClick={handleClick} className="health">Health</li>
      <li onClick={handleClick} className="local">Local</li>
      <li onClick={handleClick} className="science">Science</li>
      <li onClick={handleClick} className="technology">Technology</li>
    </ul>
  )
}

export default Menu;