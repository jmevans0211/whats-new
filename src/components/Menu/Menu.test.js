import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Menu selectTopic='local'/>);

    expect(wrapper).toMatchSnapshot();
  });

  // it('should call handleClick when clicked', () => {
  //   const wrapper = shallow(<Menu selectTopic='entertainment'/>);
  //   const handleClickMock = jest.fn()
  //   const mockEvent = {
  //     target: {
  //       className: 'entertainment'
  //     }
  //   }

  //   wrapper.find('.entertainment').simulate('click')

  //   expect(handleClickMock).toHaveBeenCalledWith(mockEvent)  
  // });

}); //<<------ end of describe block