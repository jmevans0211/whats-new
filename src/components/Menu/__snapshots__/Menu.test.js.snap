// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<ul
  role="navigation"
>
  <li
    className="entertainment"
    onClick={[Function]}
  >
    Entertainment
  </li>
  <li
    className="health"
    onClick={[Function]}
  >
    Health
  </li>
  <li
    className="local"
    onClick={[Function]}
  >
    Local
  </li>
  <li
    className="science"
    onClick={[Function]}
  >
    Science
  </li>
  <li
    className="technology"
    onClick={[Function]}
  >
    Technology
  </li>
</ul>
`;
